doc/source/first-time-contributors.rst

```diff
@@ -0,0 +1,314 @@
+and examples of `good first contributions <https://flower.dev/docs/good-first-contributions.html>`_.
+
+Setting up the repository
+-------------------------
+
+1. **Create a GitHub account and setup Git**
+    Git is a distributed version control tool. This allows for an entire codebase's history to be stored and every developper's machine.
+    It is a software that will need to be installed on your local machine, you can follow this `guide <https://docs.github.com/en/get-started/quickstart/set-up-git>`_ to set it up.
+
+    GitHub, itself, is a code hosting platform for version control and collaboration. It allows for everyone to collaborate and work from anywhere on remote repositories.
+
+    If you haven't already, you will need to create an account on `GitHub <https://github.com/signup>`_. 
+
+    The idea behind the generic Git and GitHub workflow boils down to this: 
+    you download code from a remote repository on GitHub, make changes locally and keep track of them using Git and then you upload your new history back to GitHub.
+
+2. **Forking the Flower repository**
+    A fork is a personnal copy of a GitHub repository. To create one for Flower, you must navigate to https://github.com/adap/flower (while connected to your GitHub account)
+    and click the `Fork` button situated on the top right of the page.
+
+    .. image:: _static/fork_button.png
+    
+    You can change the name if you want, but this is not necessary as this version of Flower will be yours and will sit inside your own repositories.
+    Once created, you should see on the top left corner that you are looking at your own version of Flower.
+
+    .. image:: _static/fork_link.png
+
+3. **Cloning your forked repository**
+    The next step is to download the forked repository on your machine to be able to make changes to it.
+    On your forked repository page, you should first click on the `Code` button on the right, 
+    this will give you the ability to copy the HTTPS link of the repository.
+
+    .. image:: _static/cloning_fork.png
+
+    Once you copied the \<URL\>, you can open a terminal on your machine, navigate to the place you want to download the repository to and type:
+
+    .. code-block:: shell 
+
+        $ git clone <URL>
+
+    This will create a `flower/` (or the name of your fork if you renamed it) folder in the current working directory.
+
+4. **Add origin**
+    You can then go into the repository folder:
+
+    .. code-block:: shell
+
+        $ cd flower
+
+    And here we will need to add an origin to our repository. The origin is the \<URL\> of the remote fork repository.
+    To obtain it, we can do as previously mentionned by going to our fork repository on our GitHub account and copying the link.
+
+    .. image:: _static/cloning_fork.png
+    
+    Once the \<URL\> is copied, we can type the following command in our terminal:
+
+    .. code-block:: shell
+
+        $ git remote add origin <URL>
```

Comment:
Isn't this already set when you clone from GitHub?

Reply:
Yes indeed, but it felt easier to understand the whole workflow by doing it explicitly. I can either remove the step or just mark it as optional, what do you think is best?

Reply:
Makes sense, let's leave it in there